Error handling in CG serialization

johannesduesing opened the pull request:
Up until now, we did not check the status code returned by `jvmti->GetMethodDeclaringClass`, however [it may in fact be unsuccessful](https://docs.oracle.com/javase/8/docs/platform/jvmti/jvmti.html#GetMethodDeclaringClass). This PR adds a check for the return code to indicate success before using the resulting class `cls` for further queries.

I checked this on a small sample input set for batik and results were identical to before.

There is also some changes to what i think are undesired whitespaces in `DynamicJCGAdapter.scala`, let me know if you think otherwise and i'll revert those changes.

errt reviewed (approved):
Did an error actually occur for you? I'm not sure when the universal error would be raised, but I don't think the two explicit errors mentioned should ever occur for us.

Regardless, this looks good to me.

johannesduesing commented:
When i implemented this change i got rid of the segfault on the full batik corpus - in favor of nothing happening at all, so i'm not confident i fixed the root cause there 😅 But still i figured checking the error code before proceeding with the `cls` might be a good idea, especiall since the segfault happend in a frame called `oopDesc::is_a(Klass*)` which may be something that works on the `cls` object.
